src/ontology/uberon.Makefile

```diff
@@ -1168,95 +1184,114 @@ composites: composite-metazoan.obo composite-vertebrate.obo
-		cat $@_xrefs_to_zfa.tsv | tail -n +2 | \
-		$(SCRIPTSDIR)/tbl2obolinks.pl --rel xref - > $@.tmp && mv $@.tmp $@; \
-	fi
-
-
-# Building the bridges
+# BRIDGES
 # ----------------------------------------
-# Most bridges are built here, by a script that extracts xrefs from
-# the source files prepared above and derives the bridges from them.
-$(TMPDIR)/bridges: $(TMPDIR)/seed.obo $(TMPDIR)/cl-with-xrefs.obo $(TMPDIR)/cl-zfa-xrefs.obo $(REPORTDIR)/life-cycle-xrefs.txt $(CUSTOM_BRIDGES)
-	if [ $(BRI) = true ]; then \
-		cd $(BRIDGEDIR) && \
-		perl ../../scripts/make-bridge-ontologies-from-xrefs.pl -l ../$(REPORTDIR)/life-cycle-xrefs.txt ../$(TMPDIR)/seed.obo && \
-		perl ../../scripts/make-bridge-ontologies-from-xrefs.pl -l ../$(REPORTDIR)/life-cycle-xrefs.txt -b cl ../$(TMPDIR)/cl-with-xrefs.obo ../$(TMPDIR)/cl-zfa-xrefs.obo && \
-		cd .. && touch $@; \
-	fi
+ifeq ($(BRI),true)
 
-# The above script creates OBO bridges, turn them to OWL
-$(BRIDGEDIR)/%.owl: $(BRIDGEDIR)/%.obo
-	$(OWLTOOLS) $< --remove-annotation-assertions -o $@
+# Those bridges are generated separately (see below).
+CUSTOM_BRIDGES = $(BRIDGEDIR)/uberon-bridge-to-mba.obo \
+		 $(BRIDGEDIR)/uberon-bridge-to-dmba.obo
+
+# 1. Extract cross-references from Uberon/CL and turn them to a SSSOM
+# mapping set.
+# Several cross-references are using OBO prefixes that are unknown to
+# ROBOT and not declared in Uberon/CL, so they need to be explicitly
+# declared here (otherwise xref-extract would ignore them).
+# The bridge generation pipeline is typically run with MIR=false, so
+# mirror/cl.owl may not be available; if so, we fallback to download it
+# directly from the PURL source.
+$(TMPDIR)/uberon-cl.sssom.tsv: $(SRC) mirror/cl.owl $(TMPDIR)/plugins/sssom.jar
+	if [ -f mirror/cl.owl ]; then \
+		CL_INPUT='-i mirror/cl.owl' ; \
+	else \
+		CL_INPUT='-I http://purl.obolibrary.org/obo/cl/cl-base.owl' ; \
+	fi && \
+	$(ROBOT) merge -i $< $$CL_INPUT --collapse-import-closure false \
+		 sssom:xref-extract --mapping-file $@ -v --drop-duplicates \
+		                    --prefix 'DHBA:  http://purl.obolibrary.org/obo/DHBA_'  \
+		                    --prefix 'EFO:   http://purl.obolibrary.org/obo/EFO_'   \
+		                    --prefix 'HBA:   http://purl.obolibrary.org/obo/HBA_'   \
+		                    --prefix 'KUPO:  http://purl.obolibrary.org/obo/KUPO_'  \
+		                    --prefix 'OGES:  http://purl.obolibrary.org/obo/OGES_'  \
+		                    --prefix 'PBA:   http://purl.obolibrary.org/obo/PBA_'   \
+		                    --prefix 'SCTID: http://purl.obolibrary.org/obo/SCTID_' \
+	> $(REPORTDIR)/xrefs-extraction.txt
+
+# 2. Likewise, but from ZFA (ZFA is the source of truth for the CL-ZFA
+# mappings).
+$(TMPDIR)/zfa.sssom.tsv: mirror/zfa.owl $(TMPDIR)/plugins/sssom.jar
+	if [ -f mirror/zfa.owl ]; then \
+		ZFA_INPUT='-i mirror/zfa.owl' ; \
+	else \
+		ZFA_INPUT='-I http://purl.obolibrary.org/obo/zfa/zfa-base.owl' ; \
```

Comment:
why did you choose this pattern here? How can `if [ -f mirror/zfa.owl ]` ever be false, if it is listed in the dependencies? When people run with MIR=false?

Reply:
> How can if [ -f mirror/zfa.owl ] ever be false, if it is listed in the dependencies?

That’s… a very good point! I’ll have another look, but right now I can’t see how this makes sense.

Reply:
Well, actually it does make sense.

The rule that creates the mirror uses a _shell_ conditional instead of a _Make_ conditional. That is, it basically looks like this:

```Make
mirror/zfa.owl:
    if [ $(MIR) = true ]; then ... fi
```

If we run under `MIR=false`, the mirror is not created _but_ from Make’s point of view, the rule that is supposed to create it has been executed, so the dependency is met.

So Make can trigger the execution of the bridge rule that depends on `mirror/zfa.owl` even though `mirror/zfa.owl` does not actually exist.

(This kind of situation is a big reason why I favor the use of Make conditionals, by the way.)

Reply:
> I now think it’s better and cleaner to simply make the bridge pipeline dependent on both `BRI` and `MIR`.

Nope, actually we can’t do that either, that wouldn’t be enough. In order to be sure the `mirror/zfa.owl` file exists, we need `MIR` _and_ `IMP` to be true, because the rule to create the mirror is actually dependent on both variables:

```Make
mirror/zfa.owl:
    if [ $(MIR) = true ] && [ $(IMP) = true ]; then ... ; fi
```

Re-generating the bridges only when all three variables `BRI`, `MIR`, and `IMP` are true is not an option, because that would virtually guarantee that the bridges are _never_ regenerated (the release pipeline is run with `IMP=false`).

So I’ll stick to the pattern above. It’s ugly but it works, and it can’t be make cleaner until the standard ODK Makefile is updated to stop using those shell conditionals that lead Make to think a dependency has been met even though no files has been created.

Reply:
It does mean though that we 

1. Add a potential point of failure (network dependency)
2. Add a potentially inconsistent state between bridge and MIR.



Another option could be to force create the mirror:

```
bridge.owl:
        make IMP=true MIR=true mirror/zfa.owl
        $(ROBOT)....
```

Just an idea.

Reply:
> Another option could be to force create the mirror:
> Just an idea.

Thanks, it’s still somehow a hack but I like it better than my solution.

Longer term though, I think it’s something that should be fixed at the ODK level. I don’t see why mirrors should be not created when MIR is true but IMP is false.